test cases/common/154 shared module resolving symbol in executable/meson.build

```diff
@@ -0,0 +1,24 @@
+project('shared module resolving symbol in executable', 'c')
+
+# The shared module contains a reference to the symbol 'func_from_executable',
+# which is always provided by the executable which loads it.  This symbol can be
+# resolved at run-time by an ELF loader.  But when building PE/COFF objects, all
+# symbols must be resolved at link-time, so an implib is generated for the
+# executable, and the shared module linked with it.
+#
+# See testcase 125 for an example of the more complex portability gymnastics
+# required if we do not know (at link-time) what provides the symbol.
+
+link_flags = []
+if host_machine.system() != 'windows'
+  # Needed to export dynamic symbols from the executable
+  link_flags += ['-rdynamic']
+  # Need to add this manually because Meson won't add it automatically because
+  # it doesn't know that we are loading a module from the build directory.
+  link_flags += ['-Wl,-rpath,' + meson.current_build_dir()]
```

Comment:
Don't do this, manually adding rpath entries is an easy way of getting weird bugs. We should add a new feature `build_rpath` or somesuch which is like `install_rpath` but is active in the build tree. At the same time adding manual rpath arguments should become a warning and after a few releases a hard error.